# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/Performance.h
# Commit: b1ab6c750d04
# Full Hash: b1ab6c750d04299011c1446b1752c2ff98a92b33
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-06-07 03:30:33
# Regressor Bug: 1800659
# File Overlap Count: 1
# Description:
#   Bug 1800659 - P3 Remove WorkerPrivate::ClearMainEventQueue() r=asuth,smaug
#   
#   Pending->Canceling->Killing (WorkerNeverRan)
#   
#   Need to clear WorkerPrivate::mPreStartRunnables
# ==============================================================================

diff -r 32515f9b41c7 -r b1ab6c750d04 dom/performance/Performance.h
--- a/dom/performance/Performance.h	Tue Jun 06 06:36:50 2023 +0000
+++ b/dom/performance/Performance.h	Tue Jun 06 06:36:50 2023 +0000
@@ -36,7 +36,7 @@
 class PerformanceStorage;
 class PerformanceTiming;
 class PerformanceEventTiming;
-class WorkerPrivate;
+class WorkerGlobalScope;
 class EventCounts;
 
 // Base class for main-thread and worker Performance API
@@ -52,7 +52,7 @@
       nsDOMNavigationTiming* aDOMTiming, nsITimedChannel* aChannel);
 
   static already_AddRefed<Performance> CreateForWorker(
-      WorkerPrivate* aWorkerPrivate);
+      WorkerGlobalScope* aGlobalScope);
 
   // This will return nullptr if called outside of a Window or Worker.
   static already_AddRefed<Performance> Get(JSContext* aCx,